src/sql/base/browser/ui/table/tableCellEditorFactory.ts

```diff
@@ -174,4 +175,76 @@ export class TableCellEditorFactory {
 		}
 		return TextEditor;
 	}
+
+	public getDropDownEditorClass(context: any, defaultOptions: string[]): any {
+		const self = this;
+		class TextEditor {
+			private _originalValue: string;
+			private _dropdown: Dropdown;
+			private _keyCaptureList: number[];
+
+			constructor(private _args: Slick.Editors.EditorOptions<Slick.SlickData>) {
+				this.init();
+				const keycodesToCapture = [KeyCode.Home, KeyCode.End, KeyCode.UpArrow, KeyCode.DownArrow, KeyCode.LeftArrow, KeyCode.RightArrow];
+				this._keyCaptureList = keycodesToCapture.map(keycode => getCodeForKeyCode(keycode));
+			}
+
+			/**
+			 * The text editor should handle these key press events to avoid event bubble up
+			 */
+			public get keyCaptureList(): number[] {
+				return this._keyCaptureList;
+			}
+
+			public init(): void {
+				const container = DOM.$('');
+				this._args.container.appendChild(container);
```

Comment:
You're capturing this as self above but use this here - do you actually need to capture self separately?